webapi/src/main/scala/org/knora/webapi/responders/v2/SearchResponderV2.scala

```diff
@@ -70,1066 +73,1181 @@ class SearchResponderV2 extends Responder {
 
-                                // the statement's object is a variable
+                                            val date: JulianDayNumberValueV1 = DateUtilV1.createJDNValueV1FromDateString(dateStr)
 
-                                // since all value property statements are eliminated, recreate the statement using the given predicate
+                                            // create a variable representing the period's start
+                                            val dateValueHasStartVar = createUniqueVariableNameFromEntityAndProperty(queryVar, OntologyConstants.KnoraBase.ValueHasStartJDN)
 
-                                // TODO: maybe only those value property statements have to be eliminated whose object is a literal
+                                            // sort dates by their period's start (in the prequery)
+                                            valueVariablesCreatedInFilters.put(queryVar, dateValueHasStartVar)
 
-                                AdditionalStatements(additionalStatements = Vector(
-                                    ExtendedSearchStatementPattern(subj = statementPattern.subj, pred = ExtendedSearchInternalEntityIri(OntologyConstants.KnoraBase.HasValue), obj = dateVar, false),
-                                    ExtendedSearchStatementPattern(subj = statementPattern.subj, pred = statementPattern.pred, obj = dateVar, false),
-                                    ExtendedSearchStatementPattern(subj = dateVar, pred = ExtendedSearchIri(OntologyConstants.KnoraBase.IsDeleted), obj = ExtendedSearchXsdLiteral(value = "false", datatype = OntologyConstants.Xsd.Boolean), true),
-                                    ExtendedSearchStatementPattern(subj = dateVar, pred = ExtendedSearchInternalEntityIri(OntologyConstants.Rdf.Type), obj = ExtendedSearchInternalEntityIri(OntologyConstants.KnoraBase.DateValue), true),
-                                    ExtendedSearchStatementPattern(subj = dateVar, pred = ExtendedSearchVar("dateValueObjProp" + index), obj = ExtendedSearchVar("dateValueObjVal" + index), true)
-                                ), typeInfoKeysProcessed = typeInfoKeysProcessed)
+                                            // create a variable representing the period's end
+                                            val dateValueHasEndVar = createUniqueVariableNameFromEntityAndProperty(queryVar, OntologyConstants.KnoraBase.ValueHasEndJDN)
 
-                            case _ => AdditionalStatements()
+                                            // connects the value object with the periods start variable
+                                            val dateValStartStatement = StatementPattern.makeExplicit(subj = queryVar, pred = IriRef(OntologyConstants.KnoraBase.ValueHasStartJDN), obj = dateValueHasStartVar)
 
-                   
```

Comment:
These variables are constant, you could move them out of this function into a static object.

Reply:
done in <https://github.com/dhlab-basel/Knora/pull/586/commits/22207e3eca20cb4523ae01aecc3dcbb6f11853e1>